### DISCUSSION TITLE
Auto Refactoring - move local variable as function argument with default value [Python]

### DISCUSSION THREAD

**Discussion by yamichael** [microsoft/pylance-release#5960] [Enhancement]
I'd like to suggest the following feature: when linting a local variable, constant or literal, VSCode could suggest when hitting `ctrl+.` (extract), to refactor as a function parameter, with default value as set in it.

### Example: 

**local variable to argument**

![image](https://github.com/microsoft/vscode-python/assets/24652987/1f19a454-39ed-4c95-95b2-8c132ab004ef)

**constant to argument**

![image](https://github.com/microsoft/vscode-python/assets/24652987/848adc05-1688-4387-8db1-415fe80b68af)

**current behavior** - for the example above, only _extract method_ and _extract variable_ are available:

![image](https://github.com/microsoft/vscode-python/assets/24652987/df0c7f94-93d8-4dbb-a0d6-71f66838af93)

**Comment by debonte** [comment]
Moving this issue to discussions as an enhancement request for comments and upvotes.
